### ISSUE TITLE
Global Pointer 的bug?

### ISSUE THREAD

**Issue by CheungZeeCn** [Tongjilibo/bert4torch#102]
https://github.com/Tongjilibo/bert4torch/blob/d7af4a40901e97bb47c8bb47a1decdd016645b62/bert4torch/layers.py#L961

这里面  seq _len 应该在 -3 dim 吧？ -2 应该是heads， 因为到这层的时候已经经过了前面的dense？

**Comment by CheungZeeCn** [comment]
`    def forward(self, qw, pos=None, seq_dim=-3):
        # 默认最后两个维度为[seq_len, types, hdsz]
        seq_len = qw.shape[seq_dim]
        if self.coord_dim==1:
            pos = torch.range(0, seq_len-1).long().reshape(-1)
            qw2 = torch.stack([-qw[..., 1::2], qw[..., ::2]], dim=-1).reshape_as(qw)
            # return qw * (self.cos_position[:seq_len].unsqueeze(-2)) + qw2 * (self.sin_position[:seq_len].unsqueeze(-2))
            return qw * (self.cos_position[pos].unsqueeze(-2)) + qw2 * (self.sin_position[pos].unsqueeze(-2))`

我这边类似这样写才能通

**Comment by Tongjilibo** [comment]
谢谢您的issue，这的确是个bug，您这样的实现应该也是可以的，我是在GlobalPointer里面通过转置来实现的，您试下看看是否可行
